arangod/Metrics/MetricsFeature.cpp

```diff
@@ -677,10 +652,73 @@ void MetricsFeature::toPrometheus(std::string& result, bool v2) const {
   if (engineName == RocksDBEngine::EngineName) {
     es.getStatistics(result, v2);
   }
+  if (cm.isEnabled()) {
+    cm.toPrometheus(result, _globalLabelsStr);
+  }
 }
 
 ServerStatistics& MetricsFeature::serverStatistics() noexcept {
   return *_serverStatistics;
 }
 
+constexpr auto kCoordinatorMetrics =
+    frozen::make_unordered_set<frozen::string>({
+        "arangosearch_link_stats",
+        "arangosearch_num_failed_commits",
+        "arangosearch_num_failed_cleanups",
+        "arangosearch_num_failed_consolidations",
+        "arangosearch_commit_time",
+        "arangosearch_cleanup_time",
+        "arangosearch_consolidation_time",
+    });
+
+void MetricsFeature::toVPack(VPackBuilder& builder, bool coordinator) const {
+  builder.openArray(true);
+  {
+    std::lock_guard guard(_lock);
```

Comment:
As far as I can see we need the exclusive lock only during the call to `unsafeInitTo()`, but could then go on with a shared lock to serialize the metrics.